src/NLog/Config/StackTraceUsage.cs

```diff
@@ -41,28 +41,38 @@ public enum StackTraceUsage
         /// <summary>
         /// Stack trace should not be captured.
         /// </summary>
-        None = 0, 
+        None = 0,
+
+        /// <summary>
+        /// Stack trace should only be captured if callsite details are missing
+        /// </summary>
+        WithCallSite = 1,
 
         /// <summary>
         /// Stack trace should be captured without source-level information.
         /// </summary>
-        WithoutSource = 1,
+        WithoutSource = 3,
 
 #if !SILVERLIGHT
+        /// <summary>
+        /// Stack trace (with source) should only be captured if callsite details are missing
+        /// </summary>
+        WithCallSiteSource = 2,
```

Comment:
please reoder the values. 

would it be better if we change this to a flags enum? e.g. WithCallSite | WithCallSiteSource

Reply:
StackTraceUsage is used everywhere as a level of details. The higher the number, the higher the detail. Not as a combination.

Reply:
good point, but what's higher? WithCallSiteSource  or WithoutSource ? ;)

Reply:
Also a very good point. Have now changed it into enum-flags.